products/microsoft_defender_for_endpoints.py

```diff
@@ -154,10 +161,13 @@ def nested_process_search(self, tag: Tag, criteria: dict, base_query: dict) -> N
                     if isinstance(terms, list):
                         for query_entry in terms:
                             query_entry += f" {query_base}" if query_base != '' else ''
+                            if self._limit > 0: query_entry += f"| limit {str(self._limit)}"
                             self.process_search(tag, {}, query_entry)
                     else:
                         query_entry = terms
                         query_entry += f" {query_base}" if query_base != '' else ''
+                        if self._limit > 0: query_entry += f"| limit {str(self._limit)}"
+                        
```

Comment:
This line isn't necessary since `self._limit` takes effect in the function `self.process_search`

```suggestion
```